### PR TITLE
feat: support AppConfig Deployment Strategy

### PR DESCRIPTION
**Pull request by suzuki-shunsuke**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #11973 #18032 #19307 #19320 #19324

Output from acceptance testing:

```console
$ make testacc TESTARGS='-run=TestAccAWSAppConfigDeploymentStrategy'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppConfigDeploymentStrategy -timeout 180m
=== RUN   TestAccAWSAppConfigDeploymentStrategy_basic
=== PAUSE TestAccAWSAppConfigDeploymentStrategy_basic
=== RUN   TestAccAWSAppConfigDeploymentStrategy_disappears
=== PAUSE TestAccAWSAppConfigDeploymentStrategy_disappears
=== RUN   TestAccAWSAppConfigDeploymentStrategy_Tags
=== PAUSE TestAccAWSAppConfigDeploymentStrategy_Tags
=== CONT  TestAccAWSAppConfigDeploymentStrategy_basic
=== CONT  TestAccAWSAppConfigDeploymentStrategy_Tags
=== CONT  TestAccAWSAppConfigDeploymentStrategy_disappears
--- PASS: TestAccAWSAppConfigDeploymentStrategy_disappears (45.54s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_basic (57.94s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_Tags (119.67s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	121.637s
```

Add a resource `aws_appconfig_deployment_strategy` to support [AppConfig Deployment Strategy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appconfig-deploymentstrategy.html).

### CODE REVIEW

File: aws/provider.go [hunk 1]
```diff
@@ -510,6 +510,7 @@ func Provider() *schema.Provider {
 			"aws_appconfig_configuration_profile":                     resourceAwsAppconfigConfigurationProfile(),
 			"aws_appconfig_environment":                               resourceAwsAppconfigEnvironment(),
 			"aws_appconfig_hosted_configuration_version":              resourceAwsAppconfigHostedConfigurationVersion(),
+			"aws_appconfig_deployment_strategy":                       resourceAwsAppconfigDeploymentStrategy(),
```
Comment (anGie44) [inline aws/provider.go:513]: Thanks so much for the rebase and updates @suzuki-shunsuke ! just a super quick note here to please move this line up above `environment` to keep the order

Comment (suzuki-shunsuke) [reply aws/provider.go]: Thank you for your review. Fixed 3d12fa85fcfe09ead5d7b33d1920649ef203e7c0

**Comment by suzuki-shunsuke** [comment]
I'll check the error. https://github.com/hashicorp/terraform-provider-aws/pull/19359/checks?check_run_id=3047540772

**Comment by suzuki-shunsuke** [comment]
The acceptance test was failed.
I'll check.

```console
$ make testacc TESTARGS='-run=TestAccAWSAppConfigDeploymentStrategy'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAppConfigDeploymentStrategy -timeout 180m
=== RUN   TestAccAWSAppConfigDeploymentStrategy_basic
=== PAUSE TestAccAWSAppConfigDeploymentStrategy_basic
=== RUN   TestAccAWSAppConfigDeploymentStrategy_disappears
=== PAUSE TestAccAWSAppConfigDeploymentStrategy_disappears
=== RUN   TestAccAWSAppConfigDeploymentStrategy_Tags
=== PAUSE TestAccAWSAppConfigDeploymentStrategy_Tags
=== CONT  TestAccAWSAppConfigDeploymentStrategy_basic
=== CONT  TestAccAWSAppConfigDeploymentStrategy_Tags
=== CONT  TestAccAWSAppConfigDeploymentStrategy_disappears
=== CONT  TestAccAWSAppConfigDeploymentStrategy_basic
    resource_aws_appconfig_deployment_strategy_test.go:17: Step 1/2 error: Check failed: Check 3/4 error: aws_appconfig_deployment_strategy.test: Attribute 'tags.%' expected "0", got "1"
--- FAIL: TestAccAWSAppConfigDeploymentStrategy_basic (35.20s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_disappears (48.52s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_Tags (123.25s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	125.199s
FAIL
make: *** [testacc] Error 1
```

**Comment by suzuki-shunsuke** [comment]
The acceptance test was passed.
I updated the description. https://github.com/hashicorp/terraform-provider-aws/pull/19359#issue-643941856

**Comment by anGie44** [comment]
Awesome thanks @suzuki-shunsuke ! I'll just be adding a commit to add default tags support and additional test coverage to get this last one merged in 👍

**Review by anGie44 — APPROVED** [review]
Output of acceptance test (Commercial):
```
--- PASS: TestAccAWSAppConfigDeploymentStrategy_disappears (39.48s)
--- PASS: TestAccAWSAppConfigHostedConfigurationVersion_disappears (54.24s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_disappears (55.17s)
--- PASS: TestAccAWSAppConfigEnvironment_disappears (57.36s)
--- PASS: TestAccAWSAppConfigApplication_basic (58.70s)
--- PASS: TestAccAWSAppConfigEnvironment_basic (59.00s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_basic (59.12s)
--- PASS: TestAccAWSAppConfigHostedConfigurationVersion_basic (61.47s)
--- PASS: TestAccAWSAppConfigApplication_updateName (97.24s)
--- PASS: TestAccAWSAppConfigApplication_disappears (41.93s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_updateDescription (100.44s)
--- PASS: TestAccAWSAppConfigEnvironment_MultipleEnvironments (117.42s)
--- PASS: TestAccAWSAppConfigEnvironment_updateName (121.66s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_basic (68.17s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_Validators_Multiple (70.77s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_Tags (128.94s)
--- PASS: TestAccAWSAppConfigApplication_Tags (132.47s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_updateFinalBakeTime (132.47s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_updateDescription (100.89s)
--- PASS: TestAccAWSAppConfigEnvironment_updateDescription (144.69s)
--- PASS: TestAccAWSAppConfigApplication_updateDescription (145.23s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_Validators_JSON (145.87s)
--- PASS: TestAccAWSAppConfigEnvironment_Monitors (150.84s)
--- PASS: TestAccAWSAppConfigEnvironment_Tags (150.85s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_updateName (94.64s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_Tags (103.51s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_Validators_Lambda (148.62s)
```

Output of acceptance tests (GovCloud):
```
--- PASS: TestAccAWSAppConfigDeploymentStrategy_disappears (107.54s)
--- PASS: TestAccAWSAppConfigEnvironment_disappears (170.95s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_disappears (171.09s)
--- PASS: TestAccAWSAppConfigHostedConfigurationVersion_disappears (171.31s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_basic (182.17s)
--- PASS: TestAccAWSAppConfigApplication_basic (182.19s)
--- PASS: TestAccAWSAppConfigHostedConfigurationVersion_basic (213.56s)
--- PASS: TestAccAWSAppConfigEnvironment_basic (231.07s)
--- PASS: TestAccAWSAppConfigApplication_disappears (439.04s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_basic (440.95s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_Validators_Multiple (456.08s)
--- PASS: TestAccAWSAppConfigEnvironment_updateName (628.01s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_updateDescription (633.59s)
--- PASS: TestAccAWSAppConfigApplication_Tags (635.31s)
--- PASS: TestAccAWSAppConfigEnvironment_MultipleEnvironments (637.43s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_Tags (640.97s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_updateFinalBakeTime (641.23s)
--- PASS: TestAccAWSAppConfigEnvironment_Tags (650.85s)
--- PASS: TestAccAWSAppConfigApplication_updateName (438.47s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_Tags (654.92s)
--- PASS: TestAccAWSAppConfigDeploymentStrategy_updateDescription (655.57s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_updateName (549.30s)
--- PASS: TestAccAWSAppConfigEnvironment_updateDescription (665.83s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_Validators_Lambda (502.58s)
--- PASS: TestAccAWSAppConfigEnvironment_Monitors (679.19s)
--- PASS: TestAccAWSAppConfigApplication_updateDescription (499.88s)
--- PASS: TestAccAWSAppConfigConfigurationProfile_Validators_JSON (697.76s)
```

**Pull request merged**